torch/nn/_functions/thnn/pooling.py

```diff
@@ -6,6 +6,10 @@
 from torch.nn.modules.utils import _single, _pair, _triple
 
 
+# NB: Looking for MaxPool2d or AvgPool2d?  They're natively implemented by ATen.
+# Look at tools/autograd/derivatives.yaml
+
+
 class MaxPool1d(Function):
```

Comment:
We can probably delete `MaxPool1d` and `AvgPool1d` too. They just delegate to the 2d versions in functional.py.

Reply:
What about 3d?

Reply:
Not so fast:

```
def max_pool1d(input, kernel_size, stride=None, padding=0, dilation=1,
               ceil_mode=False, return_indices=False):
    ret = _functions.thnn.MaxPool1d.apply(input, kernel_size, stride, padding, dilation,
                                          ceil_mode)
    return ret if return_indices else ret[0]

```

Reply:
Oh, I think I just forgot to enable it. We should do the same thing we're doing for avg_pool1d. Also max_pool3d should use the torch._C._nn implementation.